File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0 
        right =len(nums)-1
        List= [0] * len(nums)
        i = len(nums) -1
        
        
        while left <= right :
            if abs(nums[left]) > abs(nums[right]):
                List[i] = (nums[left] **2)          
                left+=1
            else: 
                List[i] = nums[right] ** 2
                right -=1
            i -= 1
        
        return List
                
                
                
            

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.cpp
class Solution {
public:
    vector<int> sortedSquares(vector<int>& nums) {
         int left = 0;
        int right = nums.size() - 1;
        vector<int> result(nums.size(), 0);
        int i = nums.size() - 1;

        while (left <= right) {
            if (abs(nums[left]) > abs(nums[right])) {
                result[i] = nums[left] * nums[left];
                left++;
            } else {
                result[i] = nums[right] * nums[right];
                right--;
            }
            i--;
        }

        return result;
    }
    
};